### ISSUE TITLE
Backups taking an unreasonably long time

### ISSUE THREAD

**Issue by Germar** [bit-team/backintime#365]
Using Back In Time 1.0.6 on Ubuntu 11.04, my incremental backups have gone from quick 10-15 min affairs to long, 4-6 hour, affairs.  I have disabled nice and ionice to see if that would help, but the backup seems to do a tremendous amount of time "Comparing with snapshot" and almost no time doing the new snapshot.  My current backup is about 120gigs, but that is no different from before the 11.04 upgrade.  Also, I removed all snapshots after my 11.04 upgrade and made a fresh snapshot

My most recent backup logs show no errors, just a list of changes as expected.  It is as if the compare is hanging somewhere.  Also, wasn't it the case that BiT would make the new snapshot as it went instead of doing the entire compare and then backing up?

Please see: https://answers.launchpad.net/backintime/+question/157360 for additional reports.

---

Imported from Launchpad using lp2gh.
- date created: 2011-05-18T20:14:23Z
- owner: calcipher
- the launchpad url was https://bugs.launchpad.net/bugs/784840

**Comment by Germar** [comment]
(by calcipher)
I wanted to say that I'm fairly sure I've figured out what is causing this problem.  I have many large files on my system as I have a ton of VM images, ISOs, and other things.  The huge increase in time only occurs when the 'Use checksum to detect changes' option is enabled (I think it was on by default).

I did a bit of research and rsync uses the MD4 hashing algorithm to compute these checksums.  Unfortunately, checksums on large files can take a very, very long time.  For instance, the xubuntu-11.04 iso (<1GB) tok nearly 30 seconds to compute (time openssl md4 /xubuntu-11.04-desktop-amd64.iso) and that is without nice or ionice enabled.

Turning off the checksum option causes BiT to function at a normal speed.  Perhaps having the option off should be the default option and a warning should be placed next the the option?

Note: I am unable to explain why this worked fine before 11.04.  Maybe a revision in rsync changed how the checksum was computed?
